x/yarpctest/roundtrip_test.go

```diff
@@ -0,0 +1,240 @@
+				TChannelService(
+					Name("myservice"),
+					p.Port("2"),
+					Proc(Name("echo"), EchoHandler()),
+				),
+			),
+			requests: Actions(
+				TChannelRequest(
+					p.Port("2"),
+					Body("test body"),
+					Service("myservice"),
+					Procedure("echo"),
+					ExpectRespBody("test body"),
+				),
+			),
+		},
+		{
+			name: "grpc to grpc request",
+			services: Lifecycles(
+				GRPCService(
+					Name("myservice"),
+					p.Port("3"),
+					Proc(Name("echo"), EchoHandler()),
+				),
+			),
+			requests: Actions(
+				GRPCRequest(
+					p.Port("3"),
+					Body("test body"),
+					Service("myservice"),
+					Procedure("echo"),
+					ExpectRespBody("test body"),
+				),
+			),
+		},
+		{
+			name: "response errors",
+			services: Lifecycles(
+				HTTPService(
+					Name("myservice"),
+					p.Port("4-http"),
+					Proc(
+						Name("error"),
+						ErrorHandler(
+							errors.New("error from myservice"),
+						),
+					),
+				),
+				TChannelService(
+					Name("myotherservice"),
+					p.Port("4-tch"),
+					Proc(Name("error"), ErrorHandler(errors.New("error from myotherservice"))),
+				),
+				GRPCService(
+					Name("myotherservice2"),
+					p.Port("4-grpc"),
+					Proc(Name("error"), ErrorHandler(errors.New("error from myotherservice2"))),
+				),
+			),
+			requests: Actions(
+				HTTPRequest(
+					p.Port("4-http"),
+					Service("myservice"),
+					Procedure("error"),
+					ExpectError("error from myservice"),
+				),
+				TChannelRequest(
+					p.Port("4-tch"),
+					Service("myotherservice"),
+					Procedure("error"),
+					ExpectError("error from myotherservice"),
+				),
+				GRPCRequest(
+					p.Port("4-grpc"),
+					Service("myotherservice2"),
+					Procedure("error"),
+					ExpectError("error from myotherservice2"),
+				),
+			),
+		},
+		{
+			name: "ordered requests",
+			services: Lifecycles(
+				HTTPService(
+					Name("myservice"),
+					p.Port("5"),
+					Proc(
+						Name("proc"),
+						OrderedRequestHandler(
+							ErrorHandler(yarpcerrors.InternalErrorf("internal error")),
+							StaticHandler("success"),
+							EchoHandlerWithPrefix("echo: "),
+							EchoHandler(),
+						),
+					),
+				),
+			),
+			requests: Actions(
+				HTTPRequest(
+					p.Port("5"),
+					Service("myservice"),
+					Procedure("proc"),
+					ShardKey("ignoreme"),
+					ExpectError(yarpcerrors.InternalErrorf("internal error").Error()),
+				),
+				HTTPRequest(
+					p.Port("5"),
+					Service("myservice"),
+					Procedure("proc"),
+					ExpectRespBody("success"),
+				),
+				HTTPRequest(
+					p.Port("5"),
+					Service("myservice"),
+					Procedure("proc"),
+					Body("hello"),
+					ExpectRespBody("echo: hello"),
+				),
+				HTTPRequest(
+					p.Port("5"),
+					Service("myservice"),
+					Procedure("proc"),
+					GiveAndExpectLargeBodyIsEchoed(1<<17),
+				),
+			),
+		},
+		{
+			name: "ordered request headers",
+			services: Lifecycles(
+				HTTPService(
+					Name("myservice"),
+					p.Port("6"),
+					Proc(
```

Comment:
Is there a compelling reason to abbreviate Procedure?

Reply:
It's needed to differentiate between the request option function `Procedure(..)` which specifies the procedure header and `Proc(..)` which specifies a service's handlers.